### PR TITLE
TYP: add some types to pandas/core/arrays/numpy_.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
broken off #28339

### CODE REVIEW

File: pandas/core/arrays/numpy_.py [hunk 1]
```diff
@@ -137,8 +138,8 @@ def __init__(self, values, copy=False):
         if copy:
             values = values.copy()
 
-        self._ndarray = values
-        self._dtype = PandasDtype(values.dtype)
+        self._ndarray: np.ndarray = values
```
Comment (WillAyd) [inline pandas/core/arrays/numpy_.py]: Should we move the definition to the class level as in some of the work @jbrockmendel has been doing as well? Off the top of my head I thought that was the PEP recommendation

File: pandas/core/arrays/numpy_.py [hunk 2]
```diff
@@ -137,8 +138,8 @@ def __init__(self, values, copy=False):
...
+        self._dtype: ExtensionDtype = PandasDtype(values.dtype)
```
Comment (jbrockmendel) [inline pandas/core/arrays/numpy_.py]: is PandasDtype valid here?  seems more specific

Comment (simonjayhawkins) [reply pandas/core/arrays/numpy_.py]: I'll remove for now. mypy may no longer require this type annotation.

**Review by WillAyd — APPROVED** [review]
Lgtm when green

**Comment by simonjayhawkins** [comment]
rebased + green

**Comment by WillAyd** [comment]
Thanks @simonjayhawkins

**Pull request merged**